### PR TITLE
feat: divider支持带文字配置

### PR DESCRIPTION
**Pull request by xiangwaner**
### <samp>🤖 Generated by Copilot at 94370a6</samp>

This pull request adds new features and documentation for the `divider` component in the `amis` framework, which allows users to create and customize horizontal or vertical lines that separate content. It updates the `divider` component in the `amis-ui`, `amis-editor`, and `amis` packages, as well as the corresponding documentation and CSS files. It also fixes some bugs and improves the style and usability of the component.

### <samp>🤖 Generated by Copilot at 94370a6</samp>

> _We are the dividers of the web_
> _We split the page with style and text_
> _We customize our line and label_
> _We are the dividers of the web_

### <samp>🤖 Generated by Copilot at 94370a6</samp>

*  Add the title feature to the divider component, which allows the user to add a text label to the divider and customize its style and position ([link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-58d57050233f8dd7e8a71e64c04e800d7554b9c3eb34a4578c9eab610a09c8c8R66-R69), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8R37-R64), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8R109-R115), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8L96-R162), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8L110-R226), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-efce95a42e95608f1e72013e3faefd54ae9a004f4ff27fe045949745a05607a4R2548-R2555), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-4c9fd3c4066c0163617ac92b204c4f0e4206249c1e3165717fcdd0e49a5f4d98R39-R70), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-2b4abdb9c1e1ff34f56e1905b7e5b25fe4ecc472d34817411bec72f4f926c02fR20-R22), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-2b4abdb9c1e1ff34f56e1905b7e5b25fe4ecc472d34817411bec72f4f926c02fL23-R43), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-2b4abdb9c1e1ff34f56e1905b7e5b25fe4ecc472d34817411bec72f4f926c02fL36-R56), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-2b4abdb9c1e1ff34f56e1905b7e5b25fe4ecc472d34817411bec72f4f926c02fL53-R135))
*  Modify the lineStyle and direction properties of the divider component from button-group-select to select, which is more consistent with the other properties in the editor schema ([link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8L44-R88), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8L61-R94))
*  Fix the visibleOn condition of the width and height properties of the divider component, which should use data.direction instead of direction to access the form data ([link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8L81-R121), [link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8L89-R129))
*  Add the rotate property to the divider component, which allows the user to input the rotate angle of the divider in degrees ([link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-b06ad49b8b64d0e50fe3529a2d3d5b4efe25aca142e28e03c0d9411b397134c8R109-R115))
*  Modify the CSS style of the horizontal divider to add the position and clear properties, which are needed to support the title feature of the divider ([link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-4c9fd3c4066c0163617ac92b204c4f0e4206249c1e3165717fcdd0e49a5f4d98L26-R29))
*  Modify the example code in the documentation of the divider component to show the different line styles of the divider ([link](https://github.com/baidu/amis/pull/8358/files?diff=unified&w=0#diff-58d57050233f8dd7e8a71e64c04e800d7554b9c3eb34a4578c9eab610a09c8c8L24-R55))

### CODE REVIEW

File: docs/zh-CN/components/divider.md [hunk 1]
```diff
@@ -21,11 +21,34 @@ order: 42
...
         "type": "divider",
-        "lineStyle": "solid"
+        "lineStyle": "dashed"
+    }
+]
+```
+
+## 带标题的分割线
+
+```schema: scope="body"
```
Comment (lurunze1226) [inline docs/zh-CN/components/divider.md:36]: 补充一下引入的版本号， "> `3.5.0` 及以上版本"

File: docs/zh-CN/components/divider.md [hunk 2]
```diff
@@ -40,3 +63,7 @@ order: 42
 | direction | `string` | `horizontal` | 分割线的方向，支持`horizontal`和`vertical` |
 | color     | `string` |              | 分割线的颜色                               |
 | rotate    | `number` |              | 分割线的旋转角度                           |
+| title     | [SchemaNode](../../docs/types/schemanode) |    | 分割线的标题    |
```
Comment (lurunze1226) [inline docs/zh-CN/components/divider.md:66]: 属性表加一列版本号，注明引入的版本`3.5.0`

File: docs/zh-CN/components/divider.md [hunk 3]
```diff
@@ -40,3 +63,7 @@ order: 42
 | direction | `string` | `horizontal` | 分割线的方向，支持`horizontal`和`vertical` |
 | color     | `string` |              | 分割线的颜色                               |
 | rotate    | `number` |              | 分割线的旋转角度                           |
+| title     | [SchemaNode](../../docs/types/schemanode) |    | 分割线的标题    |   
+| titleClassName | `string` |         | 分割线的标题类名                           |                    
+| orientation    | `string` | `center`| 分割线的标题位置，支持`left`、`center`和`right` |
```
Comment (lurunze1226) [inline docs/zh-CN/components/divider.md:68]: 属性名称改成`titlePosition`吧

File: packages/amis-editor/src/plugin/Divider.tsx [hunk 4]
```diff
@@ -28,6 +33,34 @@ export class DividerPlugin extends BasePlugin {
...
+              label: '带标题',
+              formType: 'extend',
+              value: false,
+              form: {
+                body: [
+                  {
+                    type: 'input-text',
+                    name: 'title',
+                    label: '标题'
+                  }
```
Comment (lurunze1226) [inline packages/amis-editor/src/plugin/Divider.tsx:54]: 这里放一个公式输入框就够了，开关没必要

**Review by lurunze1226 — CHANGES_REQUESTED** [review]

File: packages/amis-editor/src/plugin/Divider.tsx [hunk 5]
```diff
@@ -73,29 +106,62 @@ export class DividerPlugin extends BasePlugin {
...
               label: '宽度',
               name: 'style.borderWidth',
-              placeholder: '1px'
+              placeholder: '1px',
+              visibleOn: '!data.title || data.direction === "vertical"'
+            }),
+            getSchemaTpl('theme:select', {
+              mode: 'horizontal',
+              label: '宽度',
+              name: 'themeCss.titleWrapperControlClassName.border-bottom-width',
```
Comment (qkiroc) [inline packages/amis-editor/src/plugin/Divider.tsx:144]: 宽度为什么放到这个titleWrapperControlClassName下面？

**Review by qkiroc — APPROVED** [review]

**Review by lurunze1226 — APPROVED** [review]

**Pull request merged**